packages/flutter_tools/lib/src/commands/migrate_status.dart

```diff
@@ -0,0 +1,163 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:process/process.dart';
+
+import '../base/file_system.dart';
+import '../base/logger.dart';
+import '../base/platform.dart';
+import '../base/terminal.dart';
+import '../migrate/migrate_manifest.dart';
+import '../migrate/migrate_utils.dart';
+import '../project.dart';
+import '../runner/flutter_command.dart';
+import 'migrate.dart';
+
+/// Flutter migrate subcommand to check the migration status of the project.
+class MigrateStatusCommand extends FlutterCommand {
+  MigrateStatusCommand({
+    bool verbose = false,
+    required this.logger,
+    required this.fileSystem,
+    required Platform platform,
+    required ProcessManager processManager,
+  }) : _verbose = verbose,
+       migrateUtils = MigrateUtils(
+         logger: logger,
+         fileSystem: fileSystem,
+         platform: platform,
+         processManager: processManager,
+       ) {
+    requiresPubspecYaml();
+    argParser.addOption(
+      'working-directory',
+      help: 'Specifies the custom migration working directory used to stage and edit proposed changes. '
+            'This path can be absolute or relative to the flutter project root.',
+      valueHelp: 'path',
+    );
+    argParser.addOption(
+      'project-directory',
+      help: 'The root directory of the flutter project.',
+      valueHelp: 'path',
+    );
+    argParser.addFlag(
+      'diff',
+      defaultsTo: true,
+      help: 'Shows the diff output when enabled. Enabled by default.',
+    );
+    argParser.addFlag(
+      'show-added-files',
```

Comment:
what is the use case for this flag?

Reply:
Many of these flags assist in spam-reduction to make it easier to find the diffs/info the user is actually looking for.